### PR TITLE
'Using this image' drawer

### LINKED ISSUE #1109: Item page: human-readable license text
## Type
:sparkles: Feature

Building on #1078 we need to map licence codes (CC-BY etc) to plain language copy. 

## Acceptance criteria

- [x] User can view narrative statement about what licence means in practical terms: 
- under which circumstances the material can be used
- examples of permitted use

- [x] User can read a general statement about copyright of work

- [ ] User testing demonstrates that people understand their rights and obligations

- [ ] Overall reduction in calls to Enquiries teams.

## User story
As any picture researcher, I want to see clear unambiguous information about copyright and usage so that I don't get confused or sued (WI Personas research, 2014)

### PR DESCRIPTION
**Pull request by davidpmccormick**
Closes #1109

## Type
🚑 Health

- [x] Demoed to @gestchild 

## Value
Adds 'Using this image' drawer to works page

## Screenshot
screen shot 2017-11-20 at 15 51 54

### All
- [ ] Platform team to re-label 'copyright' as 'credit': https://github.com/wellcometrust/platform/pull/1192
- [x] PR labelled and assigned
- [x] Any introduced code complexity has been flagged

### CODE REVIEW

File: server/controllers/work.js [hunk 1]
```diff
@@ -31,6 +32,35 @@ function getImageIndex(ctx) {
...
+  if (creators.length > 0) {
+    const creatorsString =  creators.reduce((acc, creator, index) => {
+      if (index === 0) {
+        return `${acc} ${creator.label}`;
+      } else if (index + 1 === creators.length) {
+        return `${acc} and ${creator.label}`;
+      } else {
+        return `${acc}, ${creator.label}`;
+      }
+    }, 'by');
```
Comment (jamesgorrie) [inline server/controllers/work.js]:
```JS

creators.map(creator => creator.label).join(', ').replace(/,(?!.*,)/gmi, ', and');

```?

File: server/controllers/work.js [hunk 2]
```diff
@@ -31,6 +32,35 @@ function getImageIndex(ctx) {
...
+    return creatorsString;
+  } else {
+    return '';
+  }
+}
+
+function constructLicenseString(licenseType) {
+  const licenseInfo = getLicenseInfo(licenseType);
+  return `<a href="${licenseInfo.url}">${licenseInfo.text}</a>`;
```
Comment (jamesgorrie) [inline server/controllers/work.js]: Should we try avoid HTML in the functions and keep it in the templating?

File: server/controllers/work.js [hunk 3]
```diff
@@ -31,6 +32,35 @@ function getImageIndex(ctx) {
...
+}
+
+function constructAttribution(singleWork, canonicalUri) {
+  const title = singleWork.title ? `'${singleWork.title}' ` : '';
+  const creators = constructCreatorsString(singleWork.creators);
+  const license = constructLicenseString(singleWork.thumbnail.license.licenseType);
+  return `<p>${title} ${creators}. Source: <a href="${canonicalUri}">Wellcome Collection</a>. ${license}</p>`;
```
Comment (jamesgorrie) [inline server/controllers/work.js]: See above.

File: server/controllers/work.js [hunk 4]
```diff
@@ -48,6 +78,8 @@ export const work = async(ctx, next) => {
   const imgWidth = '800';
   const imgLink = imageUrlFromMiroId(miroId);
   const encoreLink = sierraId && encoreLinkFromSierraId(sierraId);
+  const canonicalUri = `${ctx.globals.rootDomain}/works/${singleWork.id}`;
```
Comment (jamesgorrie) [inline server/controllers/work.js]: I started removing the rootDomain as it was being duplicated on events etc.
Not sure if it's the same here.

File: server/controllers/work.js (hunk 1 above)
Comment (gestchild) [reply server/controllers/work.js]: Nice, but won't work correctly as some of the creator.label text contains commas, e.g. 'Gillray, James', which would end up as Gillray, and James

File: server/controllers/work.js (hunk 3 above)
Comment (gestchild) [reply server/controllers/work.js]: Probably yes, but not sure how to easily do this at the moment as we pass the attribution (work.attribution) to the 'work-drawer' component in the following way:

    {% set usingThisImageData = [
      {title: 'License info', content: licenseInfo.humanReadableText},
      {title: 'Attribution', content: work.attribution}
    ] %}

    {% componentV2 'work-drawer', {id: 'work-using-image', heading: 'Using this image'}, null, {data: usingThisImageData, topDivider: true} %}

I think this just highlights a problem with the component, that needs fixing.

**Comment by jennpb** [comment]
https://github.com/wellcometrust/platform/issues/1191

**Pull request merged**